### PR TITLE
Fixed the Station Map PDA app and updated Boxstation's PDA minimap

### PR DESCRIPTION
**Pull request by JMWTurner**
before
pda_minimap_box

after
pda_minimap_box

most changes were done at the north-east and south-east solars' maint tunnels. As well as the courtroom.

### CODE REVIEW

**Comment by JMWTurner** [comment]
revision 1
- Added the changes to arrivals to the minimap
- Fixed the Station Map PDA app which was actually broken ever since station Z levels are 512x512 tiles wide.

**Comment by ComicIronic** [comment]
Do we not have a tool to generate these yet?

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -1444,14 +1444,14 @@ var/global/list/obj/item/device/pda/PDAs = list()
 					if("x")
 						var/new_x = input("Please input desired X coordinate.", "Station Map App", app.markx) as num
 						var/x_validate=new_x+WORLD_X_OFFSET[map.zMainStation]
-						if(x_validate < 1 || x_validate > 255)
+						if(x_validate < (world.maxx/2 - 128) || x_validate > (world.maxx/2 + 128))
```
Comment (ComicIronic) [inline code/game/objects/items/devices/PDA/PDA.dm]: 128 should be a define as well.

**Comment by JMWTurner** [comment]
@ComicIronic We do not. The minimaps generated on server start are the crew computer's. But that gives me an idea for another PR.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -971,9 +973,9 @@ var/global/list/obj/item/device/pda/PDAs = list()
 						<img src="pda_minimap_[map.nameShort].png" style="position: relative; top: 0; left: 0;"/>
 						"}
 						if(T.z == map.zMainStation)
-							dat += {"<img src="pda_minimap_loc.gif" style="position: absolute; top: [(T.y * -1) + 247]px; left: [T.x-8]px;"/>"}
+							dat += {"<img src="pda_minimap_loc.gif" style="position: absolute; top: [(T.y * -1) + 233 + 128]px; left: [T.x + 8 - 128]px;"/>"}
```
Comment (ComicIronic) [inline code/game/objects/items/devices/PDA/PDA.dm]: You should use the define here as well.

**Comment by JMWTurner** [comment]
revision 3
- added defines where needed

@ComicIronic on the topic of automatically generated minimap, I experimented with the concept tonight and got as far as coding an holographic minimap for deathsquad agents. The map here was generated in 0.1 sec on startup. I'll try and make a similar proc for the PDA's minimaps in another PR.

hud holominimap

**Comment by clusterfack** [comment]
Ok now THAT is cool

**Comment by ihadtoregisterforthis** [comment]
After this, back to telling everyone to delete their cache for a month

**Pull request merged**